build/cmake/MongoSettings.cmake

```
@@ -0,0 +1,284 @@
+        if(DEFINED dflt_EVAL)
+            message(FATAL_ERROR "Only one of VALUE or EVAL may be specified for a DEFAULT")
+        endif()
+        set("${outvar}" "${dflt_VALUE}" PARENT_SCOPE)
+    elseif(DEFINED dflt_EVAL)
+        # Evaluate some code to determine the default
+        _mongo_eval_cmake(DEFAULT "${dflt_EVAL}")
+        set("${outvar}" "${DEFAULT}" PARENT_SCOPE)
+        if(DEFINED DEFAULT)
+            message(DEBUG "Computed default ${setting_NAME} value to be “${DEFAULT}”")
+        else()
+            message(DEBUG "No default for ${setting_NAME} was computed. Default will be an empty string.")
+        endif()
+    elseif(dflt_UNPARSED_ARGUMENTS)
+        message(FATAL_ERROR
+                "${setting_NAME}: "
+                "DEFAULT got unexpected arguments: ${dflt_UNPARSED_ARGUMENTS}")
+    endif()
+endfunction()
+
+#[==[
+Define a new boolean build setting::
+
+    mongo_bool_setting(
+        <name> <doc>
+        [DEFAULT [[DEVEL] [VALUE <value> | EVAL <code>]] ...]
+        [VALIDATE [CODE <code>]]
+        [ADVANCED] [ALLOW_OTHER_VALUES]
+    )
+
+This is a shorthand for defining a boolean setting. See mongo_setting() for more
+option information. The TYPE of the setting will be BOOL, and the implicit
+default value for the setting will be ON if no DEFAULT is provided.
+
+]==]
+function(mongo_bool_setting name doc)
+    set(args ${ARGN})
+    # Inject "ON" as a default:
+    if(NOT "DEFAULT" IN_LIST args)
+        list(APPEND args DEFAULT VALUE ON)
+    endif()
+    mongo_setting("${name}" "${doc}" TYPE BOOL ${args})
+endfunction()
+
+# Set the variable named by 'out' to the 'if_true' or 'if_false' value based on 'cond'
+function(_mongo_pick out if_true if_false cond)
+    string(REPLACE "'" "\"" cond "${cond}")
+    _mongo_eval_cmake("res" "set(res [[${if_false}]])\nif(${cond})\nset(res [[${if_true}]])\nendif()")
+    set("${out}" "${res}" PARENT_SCOPE)
+endfunction()
+
+# Evaluate CMake code <code>, and lift the given variables into the caller's scope.
+function(_mongo_eval_cmake get_variables code)
+    # Set a name that is unlikely to collide:
+    set(__eval_liftvars "${get_variables}")
+    # Clear the values before we evaluate the code:
+    foreach(__varname IN LISTS __eval_liftvars)
+        unset("${__varname}" PARENT_SCOPE)
+        unset("${__varname}")
+    endforeach()
+    # We do the "eval" the old fashion way, since we can't yet use cmake_language()
+    message(TRACE "Evaluating CMake code:\n\n${code}")
+    file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/_eval.tmp.cmake" "${code}")
+    include("${CMAKE_CURRENT_BINARY_DIR}/_eval.tmp.cmake")
+    # Lift the variables into the caller's scope
+    foreach(__varname IN LISTS __eval_liftvars)
+        if(DEFINED "${__varname}")
+            message(TRACE "Eval variable result: ${__varname}=${${__varname}}")
+            set("${__varname}" "${${__varname}}" PARENT_SCOPE)
+        endif()
+    endforeach()
+endfunction()
```

```suggestion
endfunction()

```
Missing EOF newline.